### PR TITLE
Add additional labels flag

### PR DESCRIPTION
**Pull request by rgynn**
In order to add additional parsed tags from each reasource. I've added the flag`-additional-labels`.

The flag takes a comma separated string, which is first parsed as a slice.
Then inserted into each fetcher into a slice.
The slice is later compared to the labels-map for each resource and labels are added with their corresponding value, or as an empty string if not available in resource map.

### CODE REVIEW

File: fetcher/floatingip.go
Comment (jangraefen) [inline fetcher/floatingip.go:17]: I think this could be moved to the `baseFetcher` struct, to avoid repeating it on all fetcher types, right? This obviously applies to all other occurences as well, but I am just going to mark it here 😉

File: fetcher/utils.go
Comment (jangraefen) [inline fetcher/utils.go:49]: This took me a while to figure out, mainly because I am not used to reading code with jump labels, I guess 🤔.
I know that this practice is not as frowned upon as it is in other languages and it does make the code a bit simpler.

I think for me personaly, I would prefer extracting another function for the inner loop or introduce another variables. If you find this ugly, I am also willing to getting used to having jump labels in the code base, though 😉.

**Review by jangraefen — CHANGES_REQUESTED** [review]
Hey @rgynn,

thank you so much for your contribution 🙂. I just added some minor remarks, but I think this is a very nice change! So nice of you for taking your time, adding this feature.

On last remark, even though its highly optional: If you want, you could also add some e2e tests with additional labels. If you have no interest in doing this, thats also fine. In this case, I am going to add them, after merging your PR.

**Comment by rgynn** [comment]
Hello again Jan!

Thank you so much for the code review, and that you considered to merge my PR!

I've gone ahead and made the changes you proposed, but will have to ask some help from you with the e2e tests.

Kind regards, Rikard

File: fetcher/floatingip.go
Comment (rgynn) [reply fetcher/floatingip.go]: Changed with https://github.com/jangraefen/hcloud-pricing-exporter/pull/48/commits/b443485798a87fa527b28798ca230b07afa58992

File: fetcher/utils.go
Comment (rgynn) [reply fetcher/utils.go]: Changed with https://github.com/jangraefen/hcloud-pricing-exporter/pull/48/commits/c62f5a807ae70581c6c1d69cdbfce5c8aff732de

File: fetcher/fetcher.go
Comment (jangraefen) [inline fetcher/fetcher.go:40]: I don't think it is necessary to differantiate between the `baseLabels` and the `additionalLabels` now, is it? The parameter could just be `labels ...string`, which would make all call sites alot cleaner, imo.

Its just a small refactoring, which I can also do if you want to get done with this PR, your call 🙂

File: fetcher/floatingip.go
Comment (jangraefen) [inline fetcher/floatingip.go:11]: In addition with the proposal from above, this could then become `newBase(pricing, "floatingip", "location", "type", additionalLabels...)`, which would look abit smoother, imo.

File: fetcher/utils.go
Comment (jangraefen) [inline fetcher/utils.go:43]: Very nice! At least for me, this is much clearer. I know that this might very well just be my personal taste though 😉. Thank you!

**Comment by jangraefen** [comment]
> I've gone ahead and made the changes you proposed, but will have to ask some help from you with the e2e tests.

Looks really good! I just made some additional minor remarks, based on the changes you made. 

As I said, writing e2e tests is optional and I will merge this regardless.
You can find the existing test in the `e2e` folder of the repository root. They basically create real resources by using the Hetzner API, run the fetcher under test and check, if the result is what is to be expected. Afterwards, every resources gets torn down again. Part of every test is, that the metrics produced by the fetcher have the expected labels and that calling the metric without the correct labels, gives a zero result. These test could be extended to also use the newly introduced additional labels, at least here and there, to verify everything is working correctly.
If you want to run the tests, you can do so by executing this command: `go test ./...`. It will require an environment variable called `HCLOUD_API_TOKEN` with an API token that can create resource inside your Hetzner Coud Account. This will produce some expenses, but its less then a few cents.

If you don't feel like spending your own, hard earned money, or just don't want to write tests, thats totally fine by me. It's just a couple minutes for me to add them later on, before build the release 😉

File: fetcher/fetcher.go
Comment (rgynn) [reply fetcher/fetcher.go:40]: It bredd to be differentiated in the way they are fetched from the object. The baseLabels are fetched from individual fields of the object.

The additional labels are fetched from the metadata in the labels map, using the mapping function.

Comment (jangraefen) [reply fetcher/fetcher.go:40]: I am stupid. You are so right 😄

**Review by jangraefen — APPROVED** [review]

**Comment by jangraefen** [comment]
Hey @rgynn,

just checking in: Are you still working on this or is this good to merge? I am unsure if you still doing anything or you just wait for me now? If you are all set, I will merge this ASAP.

With kind regards 🙂

**Comment by rgynn** [comment]
Hello again

Sorry for the late response. I'm done for now, you can go ahead and merge it. Thanks for the feedback!

File: fetcher/server.go
Comment (jangraefen) [inline fetcher/server.go:13]: I totally missed this one, but thankfully the test caught it 😄. This copy-paste error happend to me a few times at well... I will fix it, so no worries.

**Pull request merged**
